applications/TrilinosApplication/custom_strategies/convergencecriterias/trilinos_mixed_generic_criteria.h

```diff
@@ -183,8 +183,14 @@ class TrilinosMixedGenericCriteria : public MixedGenericCriteria< TSparseSpace,
                 dof_dx = local_dx[mpDofImport->TargetMap().LID(dof_id)];
 
                 const auto &r_current_variable = it_dof->GetVariable();
-                KeyType var_key = r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key();
-                int var_local_key = r_local_key_map[var_key];
+                const KeyType var_key = r_current_variable.IsComponent() ? r_current_variable.GetSourceVariable().Key() : r_current_variable.Key();
+                auto key_find = r_local_key_map.find(var_key);
+                if (key_find == r_local_key_map.end()) {
+                    // the dof does not belong to the list of variables
+                    // we are checking for convergence, so we skip it
+                    continue;
+                }
+                const int var_local_key = key_find->second;
```

Comment:
why not using the baseclass? Refactor to a function? :innocent:

Reply:
I put this in a function. Looking at the rest of the code I do no think there is much more you can do without overcomplicating it.